tasks/interactive-bots.md

```diff
@@ -0,0 +1,152 @@
+#GCI Tasks: interactive bots
+
+Zulip [contrib-bots](https://github.com/zulip/zulip/tree/master/contrib_bots/lib) is a boilerplate for creating interactive bots that react to messages sent by users.
+
+The interactive bots live in the `contrib-bots/lib` as `.py` files that define their specific behavior. The `contrib-bots/run.py` file defines common behaviors for the interactive bots that react to messages.
+
+The following tasks will introduce you to using interactive bots and creating simple new bots that react to messages. This group of tasks has a high creative potential, as you can create your own bots that react to specific messages and can be integrated with various APIs.
+
+## Task 0 - Test run the followup bot
+
+Follow the tutorial below to test run the followup bot on your local instance.
+
+1. Virtual environment setup:
+  a. move to the local Zulip repository
+
+  b. check out master branch in the local repository
+
+  c. pull the upstream master branch (upstream is the Zulip GitHub repository)
+
+  d. run the virtual machine using vagrant (up and provision)
+
+  e. ssh into the virtual machine
+
+  f. run migrations
+
+  ```
+  cd ~/Zulip/
+  git checkout master
+  git pull upstream master
+  vagrant up
+  vagrant provision
+  vagrant ssh
+  cd zulip
+  python manage.py migrate
+  ```
+
+2. Create a ~/`.zuliprc-local` file with credentials:
+
+  ```
+  [api]
+  key=RBY3lw7v4BNDBYMZzuBk9xkpD0Y0msIK
+  email=iago@zulip.com
+  site=http://localhost:9991/
+  ```
+
+4. Run the server using vagrant - `/srv/zulip/tools/run-dev.py`.
+
+
+5. Create a `followup` stream.
+
+
+6. Run the followup bot:
+
+  ```
+  cd ~/zulip/contrib_bots
+  python run.py lib/followup.py --config-file ~/.zuliprc-local
+  ```
+
+7. Test manually that the followup bot is working on the local Zulip instance, by:
+
+  a. sending a few messages starting with `@followup` and `@follow-up` from different streams
+
+  b. checking that the messages showed up in the `followup` stream, prepended by the sender email
```

Comment:
We need to have something for the mentors to verify for whether they're completed the task.  I guess we can ask them to post a screenshot showing it working?